src/bot/lib/query.js

```diff
@@ -11,9 +11,9 @@ export function abort(...args) {
   return error;
 }
 
-// Find matching skills for the given search term.
-export function findSkillByName(token, search) {
-  return query.skillByName({token, search}).then(matches => {
+// Find matching item (skills or categories) for the given search term.
+export function findItemByName(token, search, queryFunction) {
+  return query[queryFunction]({token, search}).then(matches => {
```

Comment:
Passing a string reference to a function is a *great* first step toward DRYing this up. Any time you find yourself doing that, consider that functions are first class citizens in JS. Why not pass the function itself? As it is now, `findItemNyName` is tightly coupled to the global scope of `query`.

As for why this matters, if you are a newcomer to this code and you see this function being used, you have to go find out how it is dereferencing the string name of the query you're handing it.

So, that would look like this:
```js
export function findItemByName(token, search, queryFunction) {
  return queryFunction({token, search}).then(matches => {
    let exact;
    if (matches.length > 0) {
      exact = matches.find(m => m.name.toLowerCase() === search.toLowerCase());
    }
    return {
      // All matches.
      matches,
      // The "best" match. Might not be exact.
      match: exact || matches[0],
      // An exact match. (case-insensitive)
      exact: exact || null,
    };
  });
}
```

...that makes it so you can pass any query function you like in, not just ones hanging off the global `query` object. Also, the callsites that use this function will now explictly require and send in everything they need to make this work.

With the previous refactor there is a now a new code "smell", you have a function immediately calling a function you handed it. Why not lift that responsibility entirely out of this function? If you look closely, the real re-usable value of this function is that it handles the results of a query. We can refactor it to do that single job very easily like so:

```js
export function parseMatches(matches) {
  let exact;
  if (matches.length > 0) {
    exact = matches.find(m => m.name.toLowerCase() === search.toLowerCase());
  }
  return {
    // All matches.
    matches,
    // The "best" match. Might not be exact.
    match: exact || matches[0],
    // An exact match. (case-insensitive)
    exact: exact || null,
  };
}
```

...now your calling code just does this:
```
query.categoryByName({token, search}).then(parseMatches);
```